### ISSUE TITLE
Errors regarding usage of deprecated core functions are too strict in a context of a namespace

### ISSUE THREAD

**Issue by markkap** [WordPress-Coding-Standards/WordPress-Coding-Standards#1244]
In the following code

```
namespace dummy;

function get_settings() {
   return 'settings';
}

echo get_settings();

```
You are going to get a notice about the usage of the deprecated function get_settings() although the one being called in this code is the "local" one.

Suggested solution, when in the context of a namespace look for the existence of a "\" before the deprecated function name.

**Comment by jrfnl** [comment]
@markkap Thanks for reporting this. A lot of sniffs have been written with PHP 5.2 support in mind and they don't all handle more modern code 100% correctly, so this will be a good example to use to fix this.

Related to #764

**Comment by markkap** [comment]
@jrfnl, yes :(

OTOH if it is impossible to know which function is being referred to, it is hard to say that you are calling a deprecated one. 

Maybe what can be done is to turn off this check when there is a namespace "use". get_settings is just an obvious generic name that should not generate errors, maybe for other deprecated functions it is less of an issue.

**Comment by jrfnl** [comment]
@markkap The error will not be turned off in WPCS itself, though I will look into checking `use` statements at some point.

In the mean time, you can turn it off for your own projects by adding `<exclude name="WordPress.WP.DeprecatedFunctions.get_settingsFound"/>` to your custom ruleset.

**Comment by markkap** [comment]
The problem is not so much with me or the other few people that right now
use WPCS and actually understand how things work and why, the problem is
the move to use it to grade plugin and theme code, and if there are too
many false positives it is a big problem, especially when it is more likely
to happen in the context of code which is compliant with the newer PHP
versions.

**Comment by jrfnl** [comment]
@markkap Methods will not generate false positives, it's just namespaced functions without the namespace prefix, so I wouldn't expect that to have much of an impact.

**Comment by markkap** [comment]
:( namespaced functions in php do not have to have any prefix. That is the
whole point of having the "use" construct. Right now WPCS will fail even if
the the RFC which suggests to eliminate the fallback to global scope will
be accepted (something that I doubt very much that will happen). It is a
false positive, you can call it too minor to matter, but it is what it is ;)

**Comment by jrfnl** [comment]
@markkap I already said, I will look into checking use statements at some point.

Other than that, you're welcome to try and find a workable solution yourself. I'd be happy to look a PR over.

**Comment by jrfnl** [comment]
Tentatively marking this for WPCS 3.0. PHPCSUtils will contain an Abstract sniff for function call detection which is better than the one we are currently using and would, at least _partially_ solve this.
